### PR TITLE
arouteserver: init at 1.23.1

### PR DESCRIPTION
**Pull request by MarcelCoding**
## Things done

This PR is already productive deployed at DD-IX.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ar/arouteserver/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    packaging
+    urllib3
+    # WHAT??? -> https://github.com/search?q=repo%3Apierky%2Farouteserver%20pkg_resources&type=code
+    setuptools
+  ];
+
+  pythonImportsCheck = [
+    "pierky"
+    "pierky.arouteserver"
+  ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/ar/arouteserver/package.nix:48]:
```suggestion
  ];

```

File: pkgs/by-name/ar/arouteserver/package.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    rev = "v${version}";
+    hash = "sha256-EZOBMDBsxbuVzzjQWU8V4n3gcLkRQxCq2eVK/Tyko4E=";
+  };
+
+  postPatch = ''
+    substituteInPlace pierky/arouteserver/builder.py --replace-fail '"bgpq4"' '"${lib.getExe bgpq4}"'
+    substituteInPlace tests/static/test_irr_queries_failover.py --replace-fail 'bgpq4 -h' '${lib.getExe bgpq4} -h'
+
+    substituteInPlace pierky/arouteserver/config/program.py --replace-fail '"bgpq4"' '"${lib.getExe bgpq4}"'
+    substituteInPlace tests/static/test_cfg_program.py --replace-fail '"bgpq4"' '"${lib.getExe bgpq4}"'
```
Comment (SuperSandro2000) [inline pkgs/by-name/ar/arouteserver/package.nix:25]:
```suggestion
    substituteInPlace tests/static/test_irr_queries_failover.py --replace-fail 'bgpq4 -h' '${lib.getExe bgpq4} -h'

    substituteInPlace pierky/arouteserver/builder.py pierky/arouteserver/config/program.py tests/static/test_cfg_program.py \
      --replace-fail '"bgpq4"' '"${lib.getExe bgpq4}"'
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/aggregate6/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  pythonImportsCheck = [ "aggregate6" ];
+
+  meta = {
+    description = "IPv4 and IPv6 prefix aggregation tool";
+    mainProgram = "aggregate6";
+    homepage = "https://github.com/job/aggregate6";
+    license = with lib.licenses; [ bsd2 ];
+    maintainers = with lib.maintainers; [ marcel ];
```
Comment (MarcelCoding) [inline pkgs/development/python-modules/aggregate6/default.nix:46]:
```suggestion
    maintainers = lib.teams.members ++ (with lib.maintainers; [ marcel ]);
```

**Comment by johannwagner** [comment]
@WDZ agrees to maintain this 🙌
We are currently using this in production, but using the packaging in our repo.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by vcunat** [comment]
```
=========================== short test summary info ============================
FAILED Copyright: is current year - AssertionError: 2024 != 2025 : Consider to run ./utils/apply_copyright
============ 1 failed, 263 passed, 25 warnings in 90.98s (0:01:30) =============
```

Excellent tests.

**Comment by MarcelCoding** [comment]
... There is already a new version out. I did not do the update myself and wanted to wait for the bot, but I guess this is not the case.

**Comment by vcunat** [comment]
Regardless of updates, tests running in nixpkgs shouldn't do such checks.  They do only harm here (spurious failures).

**Comment by MarcelCoding** [comment]
I'll work out a fix to disable specific tests.

**Comment by MarcelCoding** [comment]
I've created https://github.com/NixOS/nixpkgs/pull/371416

**Pull request merged**